File: App/Services/Microservices.Demo.Policies.Service/Domain/Policies/ValueObjects/PolicyHolder.cs
using Microservices.SharedKernel.Domain.ValueObjects;

namespace Microservices.Demo.Policies.Service.Domain.Policies.ValueObjects;

public class PolicyHolder: OfferHolder
{
    public PolicyHolder(string firstName, string lastName, string pesel, Address address)
    {
        FirstName = firstName;
        LastName = lastName;
        Pesel = pesel;
        Address = address;
    }
    protected PolicyHolder()
    {
    }
    public virtual Address Address { get; protected set; }

    protected override IEnumerable<object?> GetEqualityComponents()
    {
        yield return FirstName;
        yield return LastName;
        yield return Pesel;
        yield return Address;
    }
}

File: App/Clients/ng-microservices-demo-client/src/app/domain/policies/value-objects/policy-holder.ts
import { AutoMap } from '@automapper/classes';
import { Address } from './address';

export class PolicyHolder {
  @AutoMap()
  public firstName: string;
  @AutoMap()
  public lastName: string;
  @AutoMap()
  public pesel: string;
  @AutoMap()
  public address: Address;
  constructor(
    firstName: string,
    lastName: string,
    pesel: string,
    address: Address
  ) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.pesel = pesel;
    this.address = address;
  }
}
